docs(parachain): design for bitfield distribution

haikoschol opened the pull request:
## Changes

Add the design document for the bitfield distribution subsystem.

## Issues

#4232

haikoschol commented:
Kept as draft until we decide where under `docs` we want to put design docs.

P1sar commented:
can you please put this in to design-proposals folder?

haikoschol commented:
> can you please put this in to design-proposals folder?

According to the meeting notes, we decided on `/docs/docs/design/parachain_protocol`. Because it is already used in the docs folder (and because it looks better :) I've used kebab-case and put it in `/docs/docs/design/parachain-protocol`.

EclesioMeloJunior commented on docs/docs/design/parachain-protocol/bitfield-distribution.md:
```diff
@@ -0,0 +1,130 @@
+- messages sent to a peer for this relay parent (store the validator ID instead of the message since there can only be one per validator)
+- messages received from a peer for this relay parent (again, store validator ID, this is to avoid sending this peer a message for this validator)
+
+## Message Handling Logic
```
LGTM, just this section, I don't think that is valuable to the design if we only have the links to the function defs if we don't give any explanation about them, so maybe we can remove this section.

haikoschol replied:
I meant to write more there but then realized that I was just repeating the Rust code in prose. I think the links are useful for the person implementing it simply to find the relevant parts of the Rust code quickly.
When issues for the design are created, this section basically corresponds to issues like [this](https://github.com/ChainSafe/gossamer/issues/4313).

Instead of removing this section, I'd rather add a short high-level description of what the linked to code does.

EclesioMeloJunior replied:
Yeah, that works!

kishansagathiya reviewed (commented):
You would want to put a link to this doc in the navigation bar. you would need to modify mkdocs.yml for that.

timwu20 reviewed (approved):
LGTM! Great work @haikoschol 👍

haikoschol committed:
docs(parachain): design for bitfield distribution

haikoschol committed:
link to bitfield distribution design doc from availability subsystems page

haikoschol commented:
I've changed the branch this is based off from `feat/parachain` to `development`.

As discussed on #4349, we decided to add design documents to `development` to have them show up on https://gossamer.chainsafe.io/. The name of the feature branch is misleading now, but I figured it's the lesser evil over opening a new PR instead.

@edwardmack If you want to do the same on #4353, this is what I did:
1. checked out my feature branch locally
2. copied `bitfield-distribution.md` somewhere outside the repo
3. checked out `development`
4. renamed my feature branch (`git branch -m haiko/feat/parachain/bitfield-distribution-design haiko/feat/parachain/bak-bitfield-distribution-design`)
5. created a new feature branch with the previous name
6. copied `bitfield-distribution.md` to `docs/docs/design` and committed the added file
7. edited `availability-subsystems.md` to link to `bitfield-distribution.md` and committed the change
8. force pushed the new feature branch to github

haikoschol committed:
provide more detail on subsystem state
